blueflood-core/src/main/java/com/rackspacecloud/blueflood/service/LocatorFetchRunnable.java

```diff
@@ -81,11 +89,21 @@ public void run() {
         Set<Locator> locators = new HashSet<Locator>();
 
         try {
+            // get a list of all locators to rollup for a shard
             locators.addAll(AstyanaxReader.getInstance().getLocatorsToRollup(shard));
         } catch (RuntimeException e) {
             executionContext.markUnsuccessful(e);
             log.error("Failed reading locators for slot: " + parentSlot, e);
         }
+
+        // start a thread with EnumValidator runnable to validate enum values for this set of locators
+        try {
+            enumValidatorExecutor.execute(new EnumValidator(locators));
+        } catch (Throwable any) {
+            executionContext.markUnsuccessful(any);
```

Comment:
I don't think we want to mark the executionContext here.  it is just for rollups.  I would remove that line.

And I don't love that we are catching all Throwable's there, (I know the rollup code does, and I'm not sure it is right there either.)  I would just catch all exceptions instead of all throwables.